### ISSUE TITLE
Codeclause_Blindness Detection

### ISSUE THREAD

**Issue by Shivharekanhaiya** [Shivharekanhaiya/kanhaiya-shivhare#4]
To create an advanced graphical user interface (GUI) project for blind detection using Python and artificial intelligence (AI), we can utilize computer vision techniques and a pre-trained deep learning model. In this example, we'll use the TensorFlow and OpenCV libraries to implement the project. Please note that building a complete and accurate blindness detection system requires more extensive research and training data.

Here's an outline of the steps involved in building the GUI project:

Install the required libraries:

TensorFlow: pip install tensorflow
OpenCV: pip install opencv-python
Obtain a pre-trained model:

You need a pre-trained deep learning model for blindness detection. One popular model is MobileNetV2, trained on ImageNet. You can download it from the TensorFlow Model Zoo: https://github.com/tensorflow/models/blob/master/research/slim/nets/mobilenet/README.md
Set up the GUI using Tkinter:

Import the necessary modules and create a GUI window.
Add buttons and labels to display images and results.
Implement functions for opening images, preprocessing, and detecting blindness.
Preprocess the input image:

Load the input image selected by the user.
Resize the image to match the input size expected by the pre-trained model.
Convert the image to the appropriate format.
Perform blindness detection:

Load the pre-trained model using TensorFlow.
Pass the preprocessed image through the model.
Process the model output to determine if blindness is detected.
Display the results:

Update the GUI labels with the detection results.
Optionally, highlight the regions of the image that contribute to the blindness detection.
Here's a sample code that combines these steps to build the advanced blindness detection GUI project:

import tkinter as tk
from tkinter import filedialog
from PIL import Image, ImageTk
import cv2
import tensorflow as tf
import numpy as np

# Load the pre-trained model
model_path = 'path_to_mobilenetv2.pb'  # Update with the path to your pre-trained model
model = tf.saved_model.load(model_path)

# Define class labels
class_labels = ['Blindness', 'No Blindness']

# Function to preprocess the image
def preprocess_image(image):
    image = cv2.resize(image, (224, 224))
    image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
    image = image / 255.0  # Normalize the image
    image = np.expand_dims(image, axis=0)  # Add batch dimension
    return image

# Function to detect blindness
def detect_blindness(image):
    # Preprocess the image
    preprocessed_image = preprocess_image(image)
    
    # Perform inference
    output = model(preprocessed_image)
    
    # Get the predicted class label
    predicted_label = class_labels[np.argmax(output)]
    
    return predicted_label

# Function to open image file dialog
def open_image_dialog():
    filetypes = (("Image files", "*.jpg;*.jpeg;*.png"), ("All files", "*.*"))
    image_path = filedialog.askopenfilename(title="Select an image", filetypes=filetypes)
    
    if image_path:
        # Load and display the selected image
        image = cv2.imread(image_path)
        image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
        image.thumbnail((300, 300))  # Resize image
        photo = ImageTk.PhotoImage(Image.fromarray(image))
        image_label.configure(image=photo)
        image_label.image = photo
        
        # Detect blindness
        result = detect_blindness(image)
        result_label.configure(text=result)

# Create the main window
window = tk.Tk()
window.title("Blindness Detection")

# Create the image label
image_label = tk.Label(window)
image_label.pack(pady=10)

# Create the result label
result_label = tk.Label(window, font=("Helvetica", 16), wraplength=300)
result_label.pack(pady=10)

# Create the open image button
open_button = tk.Button(window, text="Open Image", command=open_image_dialog)
open_button.pack(pady=10)

# Start the GUI event loop
window.mainloop()

Remember to replace "path_to_mobilenetv2.pb" with the actual path to your pre-trained MobileNetV2 model.
